mage_ai/cli/main.py

```diff
@@ -1,158 +1,133 @@
-      options:
-        --runtime-vars [key value]...   specify runtime variables. These will overwrite the pipeline global variables.   
-
-    test <project_path> <pipeline_uuid> Run pipeline and output tests.
-      args:
-        project_path                    path of the Mage project that contains the pipeline.
-        pipeline_uuid                   uuid of the pipeline to be run and tested.
-      options:
-        --runtime-vars [key value]...   specify runtime variables. These will overwrite the pipeline global variables.
-
-    create_spark_cluster <project_path>
-      args:
-        project_path                    path of the Mage project that contains the EMR config.
-        """)
-    elif command == 'init':
-        from mage_ai.data_preparation.repo_manager import init_repo
-
-        repo_path = os.path.join(os.getcwd(), sys.argv[2])
-        init_repo(repo_path)
-    elif command == 'start':
-        parser = argparse.ArgumentParser()
-        parser.add_argument('repo_path', metavar='project_path', type=str)
-        parser.add_argument('--host', nargs='?', type=str)
-        parser.add_argument('--port', nargs='?', type=int)
-        parser.add_argument('--manage-instance', nargs='?', type=str)
-        parser.add_argument('--dbt-docs-instance', nargs='?', type=str)
-
-        args = dict()
-        if len(sys.argv) >= 3:
-            args = vars(parser.parse_args(sys.argv[2:]))
-            project_path = args['repo_path']
-        else:
-            project_path = os.getcwd()
-        project_path = os.path.abspath(project_path)
-
-        from mage_ai.data_preparation.repo_manager import set_repo_path
-
-        set_repo_path(project_path)
-
-        from mage_ai.server.server import start_server
-
-        start_server(
-            host=args.get('host'),
-            dbt_docs=args.get('dbt_docs_instance') == '1',
-            manage=args.get('manage_instance') == '1',
-            port=args.get('port'),
-            project=project_path,
+class OrderCommands(TyperGroup):
+    def list_commands(self, ctx: Context):
+        """Return list of commands in the order they appear."""
+        return list(self.commands)
+
+
+app = typer.Typer(
+    cls=OrderCommands,
+)
+
+
+@app.command()
+def init(
+    project_path: str = typer.Argument(..., help="path of the Mage project to be created.")
+):
+    """
+    Initialize Mage project.
+    """
+    from mage_ai.data_preparation.repo_manager import init_repo
+
+    repo_path = os.path.join(os.getcwd(), project_path)
+    init_repo(repo_path)
+    print(f"Initialized Mage project at {repo_path}")
+
+
+@app.command()
+def start(
+    project_path: str = typer.Argument(os.getcwd(), help="path of the Mage project to be loaded."),
+    host: str = typer.Option("localhost", help="specify the host."),
+    port: str = typer.Option("6789", help="specify the port."),
+    manage_instance: bool = typer.Option(False, help=""),
```

Comment:
We currently use the manage_instance command like this https://github.com/mage-ai/mage-ai-terraform-templates/blob/master/aws-dev/main.tf#L64
Updating the arg name and type will change its behaviour.

Reply:
Just reverted the types of `manage_instance` and `dbt_docs_instance` back to strings. The names should not be a problem because Typer automatically converts them to `--manage-instance` and `--dbt-docs-instance`.

![Screen Shot 2023-02-01 at 9 33 37 PM](https://user-images.githubusercontent.com/43001823/216217128-309d654b-353e-40c2-8b26-b500a87379b1.png)